src/test/java/spoon/test/trycatch/TryCatchTest.java

```diff
@@ -442,4 +444,60 @@ public void testTryWithVariableAsResource() {
 		tryStmt.removeResource(ctResource);
 
 	}
+
+	@Nested
+	class AddCatcherAt {
+		@Test
+		void addsCatcherAtTheSpecifiedPosition() {
+			// contract: the catcher should be added at the specified position
+			// arrange
+			Factory factory = createFactory();
+
+			CtTry tryStatement = factory.createTry();
+
+			CtCatch first = factory.createCatch();
+			first.setParameter(factory
+					.createCatchVariable()
+					.setType(factory.Type().createReference(IOException.class)));
+
+			CtCatch second = factory.createCatch();
+			second.setParameter(factory
+					.createCatchVariable()
+					.setMultiTypes(List.of(
+							factory.Type().createReference(ExceptionInInitializerError.class),
+							factory.Type().createReference(NoSuchFieldException.class))));
+
+			CtCatch third = factory.createCatch();
+			third.setParameter(factory
+					.createCatchVariable()
+					.setType(factory.Type().createReference(NoSuchMethodException.class)));
+
+			// act
+			tryStatement.addCatcherAt(0, third);
+			tryStatement.addCatcherAt(0, first);
+			tryStatement.addCatcherAt(1, second);
+
+			// assert
+			assertThat(tryStatement.getCatchers(), contains(first, second, third));
+		}
+
+		@Test
+		void throwsOutOfBoundsException_whenPositionIsOutOfBounds() {
+			// contract: `addCatcherAt` should throw an out-of-bounds exception when the specified position is out of
+			// bounds of the catcher collection
+
+			// arrange
+			Factory factory = createFactory();
+
+			CtTry tryStatement = factory.createTry();
+			CtCatch catcherAtWrongPosition = factory.createCatch();
+			catcherAtWrongPosition.setParameter(factory
+					.createCatchVariable()
+					.setType(factory.Type().createReference(Exception.class)));
+
+			// act & assert
+			assertThrows(IndexOutOfBoundsException.class,
+					() -> tryStatement.addCatcherAt(2, catcherAtWrongPosition));
```

Comment:
If you test only one negative case, then experience tells me that it's best to test the boundary. Boundaries are highly prone to errors (think off-by-one errors).

```suggestion
					() -> tryStatement.addCatcherAt(1, catcherAtWrongPosition));
```